crypto/x509/x509_vfy.c

```diff
@@ -1752,18 +1753,30 @@ static int internal_verify(X509_STORE_CTX *ctx)
      * is allowed to reset errors (at its own peril).
      */
     while (n >= 0) {
-        EVP_PKEY *pkey;
-
         /*
-         * Skip signature check for self signed certificates unless explicitly
-         * asked for.  It doesn't add any security and just wastes time.  If
-         * the issuer's public key is unusable, report the issuer certificate
+         * If the issuer's public key is not available or its key usage does
+         * not support issuing the subject cert, report the issuer certificate
          * and its depth (rather than the depth of the subject).
          */
+        int issuer_depth = n + (xi == xs ? 0 : 1);
+
+        if (xs != xi || sk_X509_num(ctx->chain) > 1) {
```

Comment:
I think that here just `xs != xi` is enough.  If `xs == xi` and the chain has more than one certificate, then the check would happen  again when `xs` becomes the new `xi` and we check the signature of the non-SS certificate it signed.

Reply:
@DDvO Any thoughts on the above?

Reply:
> I think that here just `xs != xi` is enough. If `xs == xi` and the chain has more than one certificate, then the check would happen again when `xs` becomes the new `xi` and we check the signature of the non-SS certificate it signed.

Correct - I've meanwhile verified this in the code, removed the `sk_X509_num(ctx->chain) > 1` check again, 
and on this occasion added a couple of comments to make the `internal_verify()` function better understandable.